eventual.go

```diff
@@ -56,11 +60,23 @@ func (v *value) Set(i interface{}) {
 		for _, waiter := range v.waiters {
 			waiter <- i
 		}
+		v.waiters = make([]chan interface{}, 0)
```

Comment:
Drive-by, we shouldn't hold on to waiters after notifying them.

Reply:
Nice catch!